packages/legend-application-studio/src/stores/LegendStudioApplicationPlugin.ts

```diff
@@ -321,4 +321,10 @@ export interface DSL_LegendStudioApplicationPlugin_Extension
    * (e.g. Class, Enum in ###Pure)
    */
   getExtraPureGrammarParserElementSnippetSuggestionsGetters?(): PureGrammarParserElementSnippetSuggestionsGetter[];
+
+  /**
+   * Get a string of the Pure grammar element name for auto-folding the element
+   * (e.g. Diagram, or Text)
+   */
+  getAutoFoldingCandidateToken?(): string[];
```

Comment:
@YannanGao-gs a small `TODO`

```suggestion
  getExtraGrammarTextEditorAutoFoldingElementCreatorKeywords?(): string[];
```

Reply:
resolved by https://github.com/finos/legend-studio/pull/2599